invenio_cli/cli.py

```diff
@@ -107,68 +53,51 @@ def init(flavour):
 @click.option('--lock/--skip-lock', default=True, is_flag=True,
               help='Lock dependencies or avoid this step')
-def build(pre, local, lock):
-    """Either setups project locally or builds container images.
+def install(pre, lock):
+    """Installs the  project locally.
 
-    --local option installs dependencies, creates instance directory,
+    Installs dependencies, creates instance directory,
     links invenio.cfg + templates, copies images and other statics and finally
     builds front-end assets.
-
-    --containers creates docker images inside of which the equivalent of
-    --local has been run.
     """
     cli_config = CLIConfig()
-    commands = Commands(cli_config, local)
-    commands.build(pre=pre, lock=lock)
+    commands = LocalCommands(cli_config)
+    commands.install(pre=pre, lock=lock)
 
 
 @cli.command()
-@click.option('--local/--containers', default=True, is_flag=True,
-              help='Which environment to build, it defaults to local')
 @click.option('--force', default=False, is_flag=True,
               help='Force recreation of db tables, ES indices, queues...')
-def services(local, force):
-    """Starts services and ensures they are setup (DB, ES, queue, etc.).
+def services(force):
+    """Starts DB, ES, queue and cache services and ensures they are setup.
 
-    TODO: Forward any extra argument to docker-compose.
+    --force destroys and resets services
     """
     cli_config = CLIConfig()
-    commands = Commands(cli_config, local)
+    commands = LocalCommands(cli_config)
     commands.services(force=force)
 
 
-# TODO: Remove when --containers implemented in the above
 @cli.command()
-@click.option('--local/--containers', default=True, is_flag=True,
-              help='Which environment to build, it defaults to local')
 @click.option('--force', default=False, is_flag=True,
-              help='Delete all content from the database, ES indexes, queues')
-@click.option('--stop-containers', default=False, is_flag=True, required=False,
-              help='Stop containers when finishing the setup operations.')
-@click.option('--verbose', default=False, is_flag=True, required=False,
-              help='Verbose mode will show all logs in the console.')
-def setup(local, force, stop_containers, verbose):
-    """Sets up the application for the first time (DB, ES, queue, etc.)."""
-    # Create config object
-    invenio_cli = InvenioCli(verbose=verbose)
-
-    click.secho('Setting up environment for {flavour} application...'
-                .format(flavour=invenio_cli.flavour), fg='green')
-
-    # Initialize docker client
-    docker_helper = DockerHelper(local=local,
-                                 log_config=invenio_cli.log_config)
+              help='Force recreation of db tables, ES indices, queues...')
+@click.option('--install-js/--no-install-js', default=True, is_flag=True,
+              help="(re-)Install JS dependencies, defaults to True")
+def containerize(force, install_js):
```

Comment:
Moderate: I could not make this run due to the lack of  `--pre` . I agree that it should not be necessary once we release. However, I imagine someone creating a package and wanting to test in containers after development.

In addition, sometimes releases are tedious and take time, so you want to have something in containers even if not fully released...

I would add the feature back, WDYT?

Reply:
Thanks for catching this: it wasn't on purpose! I had originally forgot to put locking back for `containerize` because running `invenio-cli install` locks things with `--pre` . Realizing `Pipfile.lock` would exist if `containerize` is run directly, I just added the locking back quickly without testing without prior install. I've added `pipenv lock --pre` back.